include/parser/independent/expression-compiler-clike.hpp

```diff
@@ -41,202 +50,246 @@ namespace CLikeExpressionCompilers {
 template <typename IntType>
 ExpressionCompiler<IntType> createCLikeCompiler() {
   auto binaryOperators = std::vector<ExpressionBinaryOperator<IntType>>{
```

Comment:
It never makes sense to do `auto x = Type(...)`. That's not what `auto` is good for. Just write `Type x(...)`